src/azure-cli/azure/cli/command_modules/dms/custom.py

```diff
@@ -123,60 +144,59 @@ def check_task_name_availability(client, resource_group_name, service_name, proj
                 database_options_json,
                 enable_schema_validation=False,
                 enable_data_integrity_validation=False,
                 enable_query_analysis_validation=False):
-    if os.path.exists(source_connection_json):
-        source_connection_json = get_file_json(source_connection_json)
-    else:
-        source_connection_json = shell_safe_json_parse(source_connection_json)
-
-    source_connection_info = create_sql_connection_info(source_connection_json, 'Source Database ')
-
-    if os.path.exists(target_connection_json):
-        target_connection_json = get_file_json(target_connection_json)
-    else:
-        target_connection_json = shell_safe_json_parse(target_connection_json)
 
-    target_connection_info = create_sql_connection_info(target_connection_json, 'Target Database ')
-
-    if os.path.exists(database_options_json):
-        database_options_json = get_file_json(database_options_json)
-    else:
-        database_options_json = shell_safe_json_parse(database_options_json)
-
-    database_options = []
-    for d in database_options_json:
-        database_options.append(
-            MigrateSqlServerSqlDbDatabaseInput(name=d.get('name', None),
-                                               target_database_name=d.get('target_database_name', None),
-                                               make_source_db_read_only=d.get('make_source_db_read_only', None),
-                                               table_map=d.get('table_map', None)))
-
-    validation_options = MigrationValidationOptions(enable_schema_validation=enable_schema_validation,
-                                                    enable_data_integrity_validation=enable_data_integrity_validation,
-                                                    enable_query_analysis_validation=enable_query_analysis_validation)
-
-    task_input = MigrateSqlServerSqlDbTaskInput(source_connection_info=source_connection_info,
-                                                target_connection_info=target_connection_info,
-                                                selected_databases=database_options,
-                                                validation_options=validation_options)
-
-    migration_properties = MigrateSqlServerSqlDbTaskProperties(input=task_input)
+    # Get source and target platform abd set inputs to lowercase
+    source_platform, target_platform = get_project_platforms(cmd,
+                                                             project_name=project_name,
+                                                             service_name=service_name,
+                                                             resource_group_name=resource_group_name)
+    task_type = task_type.lower()
+    scenario_handled_in_core = core_handles_scenario(source_platform,
+                                                     target_platform,
+                                                     task_type)
```

Comment:
Can `project.get` return `task_type` along with source_platform and target_platform?
- If it can return, can we use project's task_type as the default value for task creating? 
- If it can't return, can we default task_type to `""`? If passed `""`, `core_handles_scenario` can fill with default task_type for each scenario.
And `task_type` can be no longer mandatory in such ways so there will be no breaking change.

Reply:
Project task_type is currently unreliable so set to default.